docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+1. If the answer is 'yes' to question 1 or 2: are there other methods that better facilitate my use case instead of custom scripts which rely on me being a developer and not just a user? **No. Only HTTP can be used to authenticate against Nextcloud, which is not supported by Dovecot or DMS.**
+
+While it is possible to extend what Nextcloud supports with [Nextcloud apps](https://apps.nextcloud.com/), there is currently a mismatch between what DMS supports and what Nextcloud applications support. This might change in the future. For now, Lua will be used to bridge the gap between DMS and Nextcloud for authentication only (Dovecot passdb), while LDAP will still be used to identify mailboxes and verify authorization (Dovecot userdb).
+
+## Container variables to adjust
+
+Since Docker Mailserver provides minimal support for Lua with Dovecot, there are no environment variables to configure specifically for Lua. Some environment variables that do exist for other aspects of DMS might need to be configured based on the scenario you want to follow.
+
+In the case of the example scenario, the environment variables must be configured as if users will be authenticated against LDAP (to support identification of mailboxes and verifying authorizations through LDAP). See [LDAP Authentication][docs-authldap] for more information. In addition, the following variables are required:
+
+???+ example
+
+    DMS configuration environment variables to let Postfix apply Lua authentication by proxy through Dovecot for authenticated SMTP.
+
+    ```yaml
+    - ENABLE_SASLAUTHD=1
+    - SASLAUTHD_MECHANISMS=rimap
+    - SASLAUTHD_MECH_OPTIONS=127.0.0.1
+    ```
+
+## Modifying Dovecot's configuration
+
+Add the following volume values to the relevant part of `compose.yaml`:
+
+???+ example
+
+    Override and add Dovecot configuration files and lua scripts.
+
+    ```yaml
+        volumes:
+          - ./docker-data/dms/config/dovecot/auth-ldap.conf.ext:/etc/dovecot/conf.d/auth-ldap.conf.ext:ro
+          - ./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf:/etc/dovecot/conf.d/auth-lua-httpbasic.conf:ro
+          - ./docker-data/dms/config/dovecot/lua/:/etc/dovecot/lua/:ro
+    ```
+
+The first volume line [overrides][docs-dovecotoverrideconfiguration] Dovecot's standard LDAP authentication configuration file. The second line [adds][docs-dovecotaddconfiguration] a new configuration file for Lua authentication. The third line adds a directory which will contain Lua scripts. The files and directory will not be changed from inside the container, which is why they are configured as read-only.
+
+Make the necessary changes on the filesystem (*where `mailserver` is the container name*):
+```bash
+mkdir -p ./docker-data/dms/config/dovecot/lua
+docker cp mailserver:/etc/dovecot/conf.d/auth-ldap.conf.ext ./docker-data/dms/config/dovecot/auth-ldap.conf.ext
+```
```

Comment:
> * `*` should be `_` for our md italic convention.

Made this more consistent with the rest of the documentation, so the comment is not needed at all. Note that I base my documentation on other documentation pages, and some other pages also use `*` instead of `_`. It is not an issue anymore for this page.

>     * `compose.yaml` will create the lua directory itself (_although it'd be under the ownership of the user that `dockerd` runs as, which isn't always desirable_). Not necessary to hand-hold for that issue though I think.

The reason why I let the user make the directory beforehand is so that DMS does not have to be restarted (which would be overkill just to create a single directory). These files are read-only anyway. I agree that handholding is not necessary. 
 
>     * `docker cp` is fine, but this is referring to an existing container already existing.

As noted earlier, it is assumed that DMS is already up-and-running with LDAP authentication. Just referring to docker cp without any further hand-holding is fine. If a user does not realize a container needs to be running to copy files from it, the user would better not be messing around with Lua in the first place.

> When LDAP is enabled (`ACCOUNT_PROVISIONER=LDAP`), our startup scripts will disable our default Dovecot [passdb + userdb `passwd-file` config](https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/dovecot/auth-passwdfile.inc) and enable the `ldap` config instead by adjusting comments in the `10-auth.conf` config:
> 
> https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/startup/setup.d/ldap.sh#L54-L55

I am counting on this in this scenario.

> `user-patches.sh` could just as easily opt-out of the config by applying the reverse to the ldap config. Which may be relevant in future.
> 
> That has the following file contents:
> Click to view
> 
> In future for DMS we may switch to configs with separate passdb + userdb files as supporting OIDC may only be valid as a passdb AFAIK. If that happens we'll update your guide here.

That is allowed (of course!), but you can also give me a signal. I can help with the update. Lua authentication can be used for passdb + userdb, but the example only uses it for passdb. This makes it quite similar to OIDC, so it would be nice to implement it in the same way as OIDC would.
 
> Since there is anything special to copy from this config and you're advising to mount a different config anyway, there isn't much reason to include this step?

I disagree. The first line creates the lua directory so the user can work with it immediately. The second line copies the existing LDAP configuration file from the current container. This is better compared to including static configuration in the guide, in case the LDAP configuration file changes between now and when OIDC is implemented. Again, it is up to the user to make the relevant changes for the specific use case in the specific moment in time. The guide provides just an example which is kept up-to-date using best effort.

Of course, I realize an update of the guide will be necessary in the future. It'll happen when it is needed.